gitlab/_logging.py

```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+from typing import List
+
+__all__: List[str] = []
+
+# Using the `NullHandler` means that any log messages generated will not be
+# output unless the client application configures logging. For example by
+# calling `logging.basicConfig()`
+_module_root_logger_name = __name__.split(".", maxsplit=1)[0]
+logging.getLogger(_module_root_logger_name).addHandler(logging.NullHandler())
```

Comment:
Added a docstring to explain why doing it here.

Reply:
To me it just shifts the issue to imports if something happens to be imported e.g. `import _clients` would go before this. So I thought there might be a simple way for this, as I don't really see it done in popular python libraries. I know some use `get_logger()` helpers to lazily load loggers, maybe that's the idea here and I'm not getting it :grin: 

I probably don't exactly understand the risks here for our own codebase or how you stumbled across this issue, so it would just be completely subjective from my side, maybe @max-wittig or @lmilbaum would be better to take a look here :P

Reply:
@nejch Not sure I am following the discussion. Will take a deeper look at it in few hours.

Reply:
I've done some reading to figure out what the underscore means in the file name and its impact on time those lines of code are executed. Would you mind helping me figure this out?

Reply:
I think it's to indicate a private module, and it is imported here to ensure the code is evaluated before any other logging code. But I'll let John explain this :)